clients/src/main/java/org/apache/kafka/clients/consumer/MockConsumer.java

```diff
@@ -97,14 +121,47 @@ public void unsubscribe() {
     }
 
     @Override
-    public synchronized ConsumerRecords<K, V> poll(long timeout) {
+    public ConsumerRecords<K, V> poll(long timeout) {
         ensureNotClosed();
+
+        CountDownLatch pollLatchCopy = pollLatch.get();
+        if (pollLatchCopy != null) {
+            pollLatch.set(null);
+            pollLatchCopy.countDown();
+            synchronized (pollLatchCopy) {
+                // Will block until caller of waitUntilPollThen() finishes their callback.
+            }
+        }
+
+        if (wakeup.get()) {
+            wakeup.set(false);
+            throw new ConsumerWakeupException();
+        }
+
+        if (exception != null) {
+            RuntimeException exception = this.exception;
+            this.exception = null;
+            throw exception;
+        }
+
+        // Handle seeks that need to wait for a poll() call to be processed
+        for (TopicPartition tp : subscriptions.missingFetchPositions()) {
+            if (subscriptions.isOffsetResetNeeded(tp)) {
+                resetOffsetPosition(tp);
+            } else if (subscriptions.committed(tp) == null) {
```

Comment:
I didn't test this, but if I call seek(tp, long) and then poll(), without any commit, wouldn't this cause the mock consumer to reset to whatever the default strategy says and ignore my seek?